notebook-lm-clone/src/generation/rag.py

```
@@ -0,0 +1,281 @@
+Please provide a well-structured summary with proper citations:"""
+            
+            response = self.llm.call(summary_prompt)
+            
+            return RAGResult(
+                query="Document Summary",
+                response=response,
+                sources_used=sources_info,
+                retrieval_count=len(search_results)
+            )
+            
+        except Exception as e:
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)